deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.java

```diff
@@ -0,0 +1,168 @@
+package com.airbnb.deeplinkdispatch;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.TaskStackBuilder;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class BaseDeepLinkDelegate {
+
+  protected static final String TAG = "DeepLinkDelegate";
+
+  protected final List<? extends Parser> loaders;
+
+  public BaseDeepLinkDelegate(List<? extends Parser> loaders) {
+    this.loaders = loaders;
+  }
+
+  private DeepLinkEntry findEntry(String uriString) {
+    for (Parser loader : loaders) {
+      DeepLinkEntry entry = loader.parseUri(uriString);
+      if (entry != null) {
+        return entry;
+      }
+    }
+    return null;
+  }
+
+  public DeepLinkResult dispatchFrom(Activity activity) {
+    if (activity == null) {
+      throw new NullPointerException("activity == null");
+    }
+    return dispatchFrom(activity, activity.getIntent());
+  }
+
+  public DeepLinkResult dispatchFrom(Activity activity, Intent sourceIntent) {
```

Comment:
this is a nice opportunity for us to clean up/break down this method. Doesnt have to happen on this PR

Reply:
Agreed 👍 I have some requests for some new features that will be easier to work on now that I can just edit a java file.